atomic_reactor/util.py

```diff
@@ -767,6 +802,254 @@ def delete(self, relative_url, **kwargs):
         return self._do(self.session.delete, relative_url, **kwargs)
 
 
+class RegistryClient(object):
+    """
+    Registry client, provides methods for looking up image digests and configs
+    in container registries.
+
+    Methods that accept the `image` parameter expect that the registry of the
+    image matches the one of the client instance, but they do not check whether
+    that is true.
+
+    Use RegistryClient.create() to get a client for a specific registry.
+    """
+
+    def __init__(self, registry_session):
+        self._session = registry_session
+
+    @classmethod
+    def create(cls, registry, workflow, access=None):
```

Comment:
I'm not sure about this. It doesn't have additional value and it violates dependency injection principle.

IMHO sufficient is having just
```python
client = RegistryClient(RegistrySession.create_from_config(workflow, image.registry))
```

Reply:
I'll let others express about this too :)

Reply:
Any other opinions? I wouldn't mind dropping this

Reply:
Drop it please. Thanks